frontend/src/app/components/address-labels/address-labels.component.ts

```diff
@@ -46,19 +46,36 @@ export class AddressLabelsComponent implements OnInit {
         return;
       }
 
+      const topElement = this.vin.witness[this.vin.witness.length - 2];
       // https://github.com/lightning/bolts/blob/master/03-transactions.md#commitment-transaction-outputs
       if (/^OP_IF OP_PUSHBYTES_33 \w{66} OP_ELSE OP_PUSHBYTES_(1 \w{2}|2 \w{4}) OP_CSV OP_DROP OP_PUSHBYTES_33 \w{66} OP_ENDIF OP_CHECKSIG$/.test(this.vin.inner_witnessscript_asm)) {
-          if (this.vin.witness[this.vin.witness.length - 2] == '01') {
-              this.lightning = 'Revoked Force Close';
-          } else {
-              this.lightning = 'Force Close';
-          }
+        if (topElement === '01') {
+          // top element is '01' to get in the revocation path
+          this.lightning = 'Revoked Force Close';
+        } else {
+          // top element is '', this is a delayed to_local output
+          this.lightning = 'Force Close';
+        }
       // https://github.com/lightning/bolts/blob/master/03-transactions.md#offered-htlc-outputs
       } else if (/^OP_DUP OP_HASH160 OP_PUSHBYTES_20 \w{40} OP_EQUAL OP_IF OP_CHECKSIG OP_ELSE OP_PUSHBYTES_33 \w{66} OP_SWAP OP_SIZE OP_PUSHBYTES_1 20 OP_EQUAL OP_NOTIF OP_DROP OP_PUSHNUM_2 OP_SWAP OP_PUSHBYTES_33 \w{66} OP_PUSHNUM_2 OP_CHECKMULTISIG OP_ELSE OP_HASH160 OP_PUSHBYTES_20 \w{40} OP_EQUALVERIFY OP_CHECKSIG OP_ENDIF (OP_PUSHNUM_1 OP_CHECKSEQUENCEVERIFY OP_DROP |)OP_ENDIF$/.test(this.vin.inner_witnessscript_asm)) {
-        if (this.vin.witness[this.vin.witness.length - 2].length == 66) {
-            this.lightning = 'Revoked HTLC';
+        if (topElement.length === 66) {
+          // top element is a public key
+          this.lightning = 'Revoked HTLC';
```

Comment:
this can only show the word Lightning plus a message, so this gives Lightning Revoked HTLC because of
```html
<span
  *ngIf="liquid"
  class="badge badge-pill badge-warning"
  i18n="address-labels.upper-layer-peg-out"
>Liquid {{ liquid }}</span>
```

i actually want Revoked Lightning HTLC here, so i suggest not to have 3 seperate label that show up for multisig, liquid and lightning, but 1 `AddressLabelsComponent.label` that sets a label and is localized in the script, not the html. this also adds more flexibilty for future changes.

Reply:
I guess we can remove those i18n strings and do it in the script like you said. The words Lightning and Liquid in combination with the other technical terms is not viable to make translationable right now.. technical terms is often not possible to translate anyway.

Reply:
> technical terms is often not possible to translate anyway.

yes that's true

Reply:
will do it later today, i got an assignment to finish now